sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/inner/Simplify.java

```diff
@@ -418,6 +418,19 @@ public VisitDecision preorder(DBSPBinaryExpression expression) {
                     result = right;
                 }
             }
+        } else if (expression.operation.equals(DBSPOpcode.MOD)) {
+            if (right.is(DBSPLiteral.class)) {
+                DBSPLiteral rightLit = right.to(DBSPLiteral.class);
+                IsNumericType iRightType = rightType.to(IsNumericType.class);
+                if (iRightType.isOne(rightLit)) {
+                    result = iRightType.getZero();
+                } else if (iRightType.isZero(rightLit)) {
```

Comment:
Unsure as to how to check for -1. First intuition was to add `getNegativeOne` method to the `IsNumericType` interface, but it is also implemented for `usize` types.

Reply:
One way to do it is to check separately for each type.